packages/tpetra/core/src/Tpetra_Details_extractBlockDiagonal.hpp

```diff
@@ -104,6 +104,7 @@ void extractBlockDiagonal(const SparseMatrixType& A, MultiVectorType & diagonal)
         }
       }
     });
+  typename MultiVectorType::execution_space().fence();
```

Comment:
Did you actually need the fence here to avoid a crash, or was this just precautionary?

Reply:
That was a "Maybe this avoids the crash?"  Turns out it doesn't.